tools/notification-configuration/github-codeowner-subscriber/CodeOwnersParser.cs

```diff
@@ -34,17 +34,25 @@ public CodeOwnersParser(string fileContents)
         /// </summary>
         /// <param name="path">Path to search in CODEOWNERS</param>
         /// <returns>A list of contacts at the associated path or an empty list if no contacts are found</returns>
-        public List<string> GetContactsForPath(string path)
+        public List<string> GetContactsForPath(string filePath)
         {
-            var result = expressionContacts
-                .Where(expr => path.StartsWith(expr.GlobExpression))
+            var pathResults = expressionContacts
+                .Where(expr => filePath.Equals(expr.GlobExpression, StringComparison.OrdinalIgnoreCase));
+
+            // If we find an exact file path match we should not find any other fallback matches based on path
+            if (!pathResults.Any())
+            {
+                var matchPath = PathWithoutFilename(filePath);
+                var searchPath = $"/{matchPath}/";
```

Comment:
Upon re-reading my original implementation of this I think that `PathWithoutFilename` should not eliminate starting and trailing slashes. It'd be a more clean implementation. Since `PathWithoutFilename` is private and used in only one place the risk would be low.

Reply:
If we switch to only taking the last path perhaps we don't need to worry about this and we simply switch matching logic around and find the latest entry that is a prefix of yml file path. We probably want to trim any extra wildcards at the end though.